api/sys_raft.go

```diff
@@ -84,28 +86,60 @@ func (ac *AutopilotConfig) UnmarshalJSON(b []byte) error {
 
 // AutopilotState represents the response of the raft autopilot state API
 type AutopilotState struct {
-	Healthy          bool                        `mapstructure:"healthy"`
-	FailureTolerance int                         `mapstructure:"failure_tolerance"`
-	Servers          map[string]*AutopilotServer `mapstructure:"servers"`
-	Leader           string                      `mapstructure:"leader"`
-	Voters           []string                    `mapstructure:"voters"`
-	NonVoters        []string                    `mapstructure:"non_voters"`
+	Healthy                    bool                        `mapstructure:"healthy"`
+	FailureTolerance           int                         `mapstructure:"failure_tolerance"`
+	Servers                    map[string]*AutopilotServer `mapstructure:"servers" deep:"-"`
+	Leader                     string                      `mapstructure:"leader"`
+	Voters                     []string                    `mapstructure:"voters"`
+	NonVoters                  []string                    `mapstructure:"non_voters"`
+	RedundancyZones            map[string]AutopilotZone    `mapstructure:"redundancy_zones,omitempty"`
+	Upgrade                    *AutopilotUpgrade           `mapstructure:"upgrade_info,omitempty"`
+	OptimisticFailureTolerance int                         `mapstructure:"optimistic_failure_tolerance,omitempty"`
 }
 
 // AutopilotServer represents the server blocks in the response of the raft
 // autopilot state API.
 type AutopilotServer struct {
-	ID          string            `mapstructure:"id"`
-	Name        string            `mapstructure:"name"`
-	Address     string            `mapstructure:"address"`
-	NodeStatus  string            `mapstructure:"node_status"`
-	LastContact string            `mapstructure:"last_contact"`
-	LastTerm    uint64            `mapstructure:"last_term"`
-	LastIndex   uint64            `mapstructure:"last_index"`
-	Healthy     bool              `mapstructure:"healthy"`
-	StableSince string            `mapstructure:"stable_since"`
-	Status      string            `mapstructure:"status"`
-	Meta        map[string]string `mapstructure:"meta"`
+	ID             string `mapstructure:"id"`
+	Name           string `mapstructure:"name"`
+	Address        string `mapstructure:"address"`
+	NodeStatus     string `mapstructure:"node_status"`
+	LastContact    string `mapstructure:"last_contact"`
+	LastTerm       uint64 `mapstructure:"last_term"`
+	LastIndex      uint64 `mapstructure:"last_index"`
+	Healthy        bool   `mapstructure:"healthy"`
+	StableSince    string `mapstructure:"stable_since"`
+	Status         string `mapstructure:"status"`
+	Version        string `mapstructure:"version"`
+	UpgradeVersion string `mapstructure:"upgrade_version,omitempty"`
+	RedundancyZone string `mapstructure:"redundancy_zone,omitempty"`
+	ReadReplica    bool   `mapstructure:"read_replica,omitempty"`
```

Comment:
I would like to avoid exposing the ReadReplica terminology to users.  I mean I'd like to avoid seeing it in the code at all but I realize that's likely not feasible.  Anyway I'm not sure that this is user-facing, if not we can resolve this.

Reply:
This is user facing, in that it's an API response. I can remove it.